### PR TITLE
Debian container for VS Codespaces

### PR DESCRIPTION
**Pull request by vhvb1989**
Provide a Docker file to spin up the dev-env for developers so they don't need to make a local set up.

fixes: https://github.com/Azure/azure-sdk-for-cpp/issues/1920

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+ARG USERNAME=azure-sdk-for-cpp
+ARG USER_UID=1000
+ARG USER_GID=$USER_UID
+ARG PORT=4000
+
+# Install packages as root
+USER root
+
+# Install needed packages and setup non-root user. Use a separate RUN statement to add your own dependencies.
+RUN apt-get update \
```
Comment (RickWinter) [inline .devcontainer/Dockerfile:16]: Although its convenient for size of the image (reduced layers) to run everthing as 1 command its hard to debug when it fails.  You might consider breaking this into a couple RUN steps broken up by purposes "Create user" "Install packages" "Setup\Cleanup Environment and temp files"

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -42,9 +42,18 @@ Merging Pull Requests (for project contributors with write access)
   - Contributor is using an e-mail address other than the primary GitHub address and wants that preserved in the history. Contributor must be willing to squash
     the commits manually before acceptance.
 
+## Codespaces
```
Comment (RickWinter) [inline CONTRIBUTING.md:45]: Leave ## Developer Guide as the top level.
Put ### Codespaces and ### VS Code Codespaces under that topic

File: .devcontainer/Dockerfile (hunk 1 above)
Comment (vhvb1989) [reply .devcontainer/Dockerfile:16]: I tried by `RUN` will be executed in parallel to any other `RUN` and we need sequential execution here.

**Review by antkmsft — APPROVED** [review]

**Comment by ahsonkhan** [comment]
Make sure to run the link verification step in PRs like these to ensure CI remains green:
image

**Pull request merged**